src/dso_api/dynamic_api/serializers.py

```diff
@@ -130,18 +100,20 @@ class DynamicListSerializer(DSOModelListSerializer):
     """
 
     def get_prefetch_lookups(self) -> list[Union[models.Prefetch, str]]:
-        """Optimize M2M prefetch lookups to return latest temporal records only."""
+        """Optimize M2M prefetch lookups to return correct temporal records only."""
         parent_model = self.child.Meta.model
         lookups = super().get_prefetch_lookups()
+        request = self.context["request"]
 
         for i, lookup in enumerate(lookups):
             related_model, is_many = resolve_model_lookup(parent_model, lookup)
             if cast(DynamicModel, related_model).is_temporal() and is_many:
 
-                # When the model is a temporal relationship, make sure the prefetch only
-                # returns the latest objects. The Prefetch objects allow adding these filters.
+                # When the model is a temporal relationship, make sure the prefetch only returns
+                # the proper temporal slice. The Prefetch objects allow adding these filters.
                 lookups[i] = models.Prefetch(
-                    lookup, queryset=filter_latest_temporal(related_model.objects)
+                    lookup,
+                    queryset=filter_temporal_slice(request, related_model.objects.all()),
```

Comment:
You already know you are dealing with a temporal model (see line 110). Why should `filter_temporal_slice` check that again? Maybe you should simply inline the call to `TemporalTableQuery`

Reply:
The `filter_latest_temporal()` is called from many other places, and doesn't require some pre-assert check before being called. In this situation it's a double check, but on other code paths it makes things much simpler.